File: 2024-11-November-GFG-POTD/November 11 - Make Array Elements Unique/Solution.cpp
// { Driver Code Starts
// Initial Template for C++

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int minIncrements(vector<int>& arr) {
        // Sort the array first
        sort(arr.begin(), arr.end());
        
        // Initialize increment count
        int increments = 0;
        
        // Iterate through the sorted array
        for (int i = 1; i < arr.size(); i++) {
            // If the current element is not greater than the previous element,
            // increment it to make it unique
            if (arr[i] <= arr[i - 1]) {
                int requiredIncrement = arr[i - 1] + 1 - arr[i];
                arr[i] = arr[i - 1] + 1; // Update the current element
                increments += requiredIncrement;
            }
        }
        
        return increments;
    }
};

int main() {
    int t;
    cin >> t;
    while (t--) {
        string input;
        cin.ignore();
        getline(cin, input);
        istringstream ss(input);
        vector<int> arr;
        int num;
        while (ss >> num) {
            arr.push_back(num);
        }
        Solution obj;
        int result = obj.minIncrements(arr);
        cout << result << endl;
    }
    return 0;
}
// } Driver Code Ends

File: 2024-11-November-GFG-POTD/November 11 - Make Array Elements Unique/Solution.py
# { Driver Code Starts
# Initial Template for Python

class Solution:
    def minIncrements(self, arr):
        # Sort the array first
        arr.sort()
        
        # Initialize increment count
        increments = 0
        
        # Iterate through the sorted array
        for i in range(1, len(arr)):
            # If the current element is not greater than the previous element,
            # increment it to make it unique
            if arr[i] <= arr[i - 1]:
                required_increment = arr[i - 1] + 1 - arr[i]
                arr[i] = arr[i - 1] + 1 # Update the current element
                increments += required_increment
        
        return increments

# Driver Code
if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        arr = list(map(int, input().split()))
        obj = Solution()
        result = obj.minIncrements(arr)
        print(result)
# } Driver Code Ends

